server/common/rest.py

```diff
@@ -222,6 +222,10 @@ def data_var_get(request, data_adaptor):
         return abort_and_log(HTTPStatus.BAD_REQUEST, str(e), include_exc_info=True)
 
 
+def colors_get(data_adaptor):
+    return make_response(jsonify(data_adaptor.colors()), HTTPStatus.OK)
```

Comment:
error handling?  What if `colors()` fails?  Right now that will generate a 500 response, which seems brittle.

Suggest that it either return an empty color dictionary, OR return a useful HTTP error (eg, 404 or something like that)

Reply:
Both methods currently return an empty dictionary if there is no color information in the file. I will add a `try` here, though I'm not entirely sure the extent of exceptions that could be thrown here!

Reply:
I am more worried about badly formatted color info.  Eg., user hand created, etc.

Reply:
I can provide an example if you want one - there are lots of things that will potentially cause this to blow up.

Reply:
> I am more worried about badly formatted color info

I _think_ f1b2f0b addresses the core concern here.

WRT other color failure modes, the only other case I can think of is where the user puts in a `adata.uns['{category_name}_colors']` key for a category that does not exist in `adata.obs`. The current code will simply leave this out of the cxg color data structure, which I think is better behavior than throwing an exception.